docs/Architecture Decision Record/012-antivirus-strategy.md

```diff
@@ -0,0 +1,41 @@
+# 12. Anti-virus Strategy Change
+
+Date: 2021-04-01 (yyyy-mm-dd)
+
+## Status
+
+Accepted
+
+## Context
+
+Our original plan was to upload files to S3 using signed URLs from S3, and checking the files' headers to determine whether or not it was, in fact, a text file. Further research has revealed that there is no fool proof way of doing this and a malicious user would be able to spoof a virus as a text file.
+
+## Decision
+
+Instead of using a signed URL and sending the file directly to S3, we will instead send the file to the backend and scan it with Clam AV before sending it to S3. In the event there is a virus, we will destroy the file on the backend immediately and return an error to the frontend. 
+
+In addition to this, the frontend is able to reliably determine if a file is a binary file. The client will check submitted files for this and immediately return an error to the user. These files will not be sent to the backend.
```

Comment:
The ClamAV decision looks great to me! As we build we may discover that some of the architectural implementation details will need to change and that's OK.